internal/initializer/crds.go

```diff
@@ -131,6 +135,35 @@ func (c *CoreCRDs) Run(ctx context.Context, kube client.Client) error {
 		if err := pa.Apply(ctx, crd); err != nil {
 			return errors.Wrap(err, "cannot apply crd")
 		}
+
+		crdNames = append(crdNames, crd.Name)
 	}
+
+	// wait for crds to be established
+	pollInterval := 2 * time.Second
+	timeout := 60 * time.Second
+	for _, crdName := range crdNames {
+		if err := wait.PollUntilContextTimeout(ctx, pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
why not switch the order? Seems to be easier to reason about. I.e. one poll and a loop inside waiting for the condition.

Reply:
Similar to what @phisco [commented](https://github.com/crossplane/crossplane/pull/5949#discussion_r1752421244), my pov was to not keep common waiting period for all crds as it might lead to increasing this waiting period with the increasing number of CRD and will again need a code change here.

Or let's keep 2 minute as common timeout with one poll outside the loop?

Reply:
it's probably even desirable to have to change this code here if we ever end up needing more time, that would mean  there is a bigger problem and we should probably reconsider our approach altogether. Otherwise we'd be hiding it. 2 mins sounds reasonable

Reply:
Sure @phisco, makes sense. I've added changes to wait for 2 minutes for all CRDs.

I've used map[string]struct{} for mainitaining pending CRDs as a struct{} takes 0 bytes of memory and map offers O(1) deletion.